### DISCUSSION TITLE
Translation-API is not working on mobile devices

### DISCUSSION THREAD

**Discussion by AstridOttmann** [neuefische/questions#522] [Web] [answered]
The implemented translation feature (deepl-API) is working on the mac so far, but when I open it on my smartphone, I get no translation at all. 
What is the problem? How can I change that? Here's the link [branch speech&translation](https://github.com/AstridOttmann/capstone-project/pull/25)

**Comment by ThomasF85** [comment, marked as answer]
Hi Astrid,

when I open your preview deployment the translation feature doesn't work. The problem seems to be the preview deployment. You get a server error when calling `/api/deepl`.
This API enpoint of yours uses the following: `process.env.DEEPL_AUTH_KEY`
You have to set the environment variable DEEPL_AUTH_KEY in Vercel (and give it the same value it has in your .env.local file): https://vercel.com/docs/concepts/projects/environment-variables

**Reply by AstridOttmann** [reply]
Thank you for your answer. I forgot this. 
Now I did it, but still don't get it working on mobile. Is it working with you or do I have to make a new PR?

**Reply by pixelass** [reply]
you have to trigger a new deployment. either update your PR or trigger the last deployment on vercel:
![react-stickyroll_–_Deployments_-_Vercel](https://user-images.githubusercontent.com/1148334/218091114-21a80b00-562b-40d9-ad4f-9ffb0cba0807.png)

**Reply by AstridOttmann** [reply]
Thanks. I redeployed it, but still don't get the translation.

**Reply by pixelass** [reply]
can you check if there is an error log in vercel and post the log  here?

When the functions page is open you have to open the client (the app in the browser) once the page where the function is called you should see something happening in the function log.

If you don't navigate to that page the logs will remain empty since they are LIVE logs


![react-stickyroll_–_Overview_-_Vercel00](https://user-images.githubusercontent.com/1148334/218095253-d27623ac-6df3-4772-8a15-ee266c061a74.png)

![react-stickyroll_–_Functions00](https://user-images.githubusercontent.com/1148334/218095252-381d22ed-a504-403d-9c94-f21592f70024.png)

**Reply by AstridOttmann** [reply]
Did you mean it like this? 
![Screenshot 2023-02-10 at 2 38 43 PM](https://user-images.githubusercontent.com/118136613/218105970-2fecec9d-3351-4a12-9849-2b5d80fba019.png)
I don't get any logs on vercel, only in the console

**Reply by ThomasF85** [reply]
The problem has been solved by now. We were looking at the wrong deployment logs on Vercel and could therefore not see the function logs at first.
The problem was ultimately solved by:
1. Adding the DEEPL_AUTH_KEY environment variable on Vercel
2. Making sure only one response per API request was being send.
